interpreter/runtimecost.go

```diff
@@ -170,19 +188,60 @@ func (c CostTracker) actualSize(value ref.Val) uint64 {
 	return 1
 }
 
+type stackVal struct {
+	Val ref.Val
+	ID  int64
+}
+
 // refValStack keeps track of values of the stack for cost calculation purposes
-type refValStack []ref.Val
+type refValStack []stackVal
 
-func (s *refValStack) push(value ref.Val) {
+func (s *refValStack) push(value stackVal) {
 	*s = append(*s, value)
 }
 
-func (s *refValStack) pop(count int) ([]ref.Val, bool) {
-	if len(*s) < count {
-		return nil, false
+// drop1 removes the top item from the stack.
+func (s *refValStack) drop1() {
+	if len(*s) < 1 {
+		return
 	}
-	idx := len(*s) - count
-	el := (*s)[idx:]
+	idx := len(*s) - 1
 	*s = (*s)[:idx]
-	return el, true
+}
+
+// TODO: Allowing drop and popArgs to remove stack items above the IDs they are provided is a workaround. drop and popArgs
+// should find and remove only the stack items matching the provided IDs once all attributes are properly pushed and popped from stack.
+
+// drop searches the stack for each ID and removes the ID and all stack items above it.
+// If none of the IDs are found, the stack is not modified.
+func (s *refValStack) drop(ids ...int64) {
```

Comment:
It's possible the ids are reported more than once with the way `Attribute` values are constructed and the way macros manipulate the AST. Macros should generally not do this, but there's no auditing to ensure they don't. It's worth updating the comment to indicate these are possible causes.

Reply:
I've added a warning.I tried to reason through if this would have any problem implications and couldn't think of any.